packages/data/src/components/use-select/index.js

```diff
@@ -248,6 +254,8 @@ export default function useSelect( mapSelect, deps ) {
 		// examining every individual value inside the `deps` array.
 	}, [ registry, wrapSelect, hasMappingFunction, depsChangedFlag ] );
 
+	useDebugValue( mapOutput );
```

Comment:
Is there any recommended way to ensure it doesn't run on production?

Reply:
we should be safe to break the rule of hooks based on a compile-time constant like `NODE_ENV === 'production'` because the number of hooks would never change within a running session. we'd need a comment and an ignore statement indicating as much though.

Reply:
It looks like it's covered by React internally:

https://github.com/facebook/react/blob/dd2d6522754f52c70d02c51db25eb7cbd5d1c8eb/packages/react/src/ReactHooks.js#L150-L158

I see that it's a common way to consume this hooks by libraries.

Jotai:

https://github.com/pmndrs/jotai/blob/a94a0ebb6e27365e5b018eff0061edcda0670179/src/core/useAtomValue.ts#L88

Zustand:

https://github.com/pmndrs/zustand/blob/3d764b0a4a0cb5f785b32673a97cf149381acf6d/src/react.ts#L41

Reply:
noting that the solution to rely on React internally leaves in the function call while wrapping with our own `NODE_ENV` check should strip it out of the production build entirely.